<a href="https://colab.research.google.com/github/JFaberSFSD/Healthcare_Predictions/blob/main/Data_Option_1_Heart_Disease.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [1]:
import pandas as pd

In [2]:
df = pd.read_csv('/content/drive/MyDrive/07 - Week 7 (August 1st)/heart.csv')

df.head()

Age Sex ChestPainType  RestingBP  Cholesterol  FastingBS RestingECG  MaxHR  \
0   40   M           ATA        140          289          0     Normal    172   
1   49   F           NAP        160          180          0     Normal    156   
2   37   M           ATA        130          283          0         ST     98   
3   48   F           ASY        138          214          0     Normal    108   
4   54   M           NAP        150          195          0     Normal    122   

  ExerciseAngina  Oldpeak ST_Slope  HeartDisease  
0              N      0.0       Up             0  
1              N      1.0     Flat             1  
2              N      0.0       Up             0  
3              Y      1.5     Flat             1  
4              N      0.0       Up             0

In [3]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 918 entries, 0 to 917
Data columns (total 12 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   Age             918 non-null    int64  
 1   Sex             918 non-null    object 
 2   ChestPainType   918 non-null    object 
 3   RestingBP       918 non-null    int64  
 4   Cholesterol     918 non-null    int64  
 5   FastingBS       918 non-null    int64  
 6   RestingECG      918 non-null    object 
 7   MaxHR           918 non-null    int64  
 8   ExerciseAngina  918 non-null    object 
 9   Oldpeak         918 non-null    float64
 10  ST_Slope        918 non-null    object 
 11  HeartDisease    918 non-null    int64  
dtypes: float64(1), int64(6), object(5)
memory usage: 86.2+ KB


In [4]:
df['HeartDisease'].value_counts()

1    508
0    410
Name: HeartDisease, dtype: int64

#1) Source of data
> ### Kaggle (via Discord Post)

#2) Brief description of data
> ### Data from subjects who may or may not have heart disease. We have 10 different columns of data about each patient to help our model
#3) What is the target?
> ### The target is the 'HeartDisease' column, 1 for 'yes' and 0 for 'no'
#4) Is this a classification or regression problem?
> ### Classification for sure
#5) How many features?
> ### 10 features, with the 11th column being our target
#6) How many rows of data.
> ### There are 918 rows of data
#7) What, if any, challenges do your foresee in cleaning, exploring, or modeling with this dataset?
> ### Overall this isn't a ton of data to work with, just 918 rows. What makes this my number 1 choice over stroke though, is the balance of the dataset. It is fairly close to a 50/50 split across our data, which should help our model overall. Even though this dataset is 1/5th the size of the stroke dataset, we actually have DOUBLE the amount of positive cases